samples/net/mqtt_publisher/overlay-bt.conf

```diff
@@ -0,0 +1,9 @@
+CONFIG_BT=y
+CONFIG_BT_DEBUG_LOG=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_L2CAP_DYNAMIC_CHANNEL=y
+CONFIG_BT_DEVICE_NAME="Zephyr MQTT"
+CONFIG_NET_L2_BT=y
+CONFIG_NET_IPV4=n
+CONFIG_NET_IPV6=y
```

Comment:
Following suggestion from @pfalcon: Adding `CONFIG_NET_CONFIG_NEED_IPV6=y` will make the sample wait until BLE connection is established.